features/mbedtls/targets/TARGET_STM/md5_alt.c

```diff
@@ -90,47 +90,49 @@ void mbedtls_md5_clone( mbedtls_md5_context *dst,
     *dst = *src;
 }
 
-void mbedtls_md5_starts( mbedtls_md5_context *ctx )
+int mbedtls_md5_starts_ret( mbedtls_md5_context *ctx )
 {
     /* HASH IP initialization */
     if (HAL_HASH_DeInit(&ctx->hhash_md5) != 0) {
-        // error found to be returned
-        return;
+        return MBEDTLS_ERR_MD5_HW_ACCEL_FAILED;
     }
 
     /* HASH Configuration */
     ctx->hhash_md5.Init.DataType = HASH_DATATYPE_8B;
     /* clear CR ALGO value */
     HASH->CR &= ~HASH_CR_ALGO_Msk;
     if (HAL_HASH_Init(&ctx->hhash_md5) != 0) {
-        // return error code
-        return;
+        return MBEDTLS_ERR_MD5_HW_ACCEL_FAILED;
     }
     if (st_md5_save_hw_context(ctx) != 1) {
-        return; // return HASH_BUSY timeout Error here
+        return MBEDTLS_ERR_MD5_HW_ACCEL_FAILED; // Hash busy timeout
```

Comment:
A general error ```MBEDTLS_ERR_MD5_HW_ACCEL_FAILED``` here hides the timeout error 'HASH_BUSY' that could simply be to indicate that a retry after a delay may succeed. In general we have to preserve status indications from HW that do not mean actual failure. This should be reviewed by original developer to confirm if error code is correct.

Reply:
The HW_ACCEK_FAILED codes have been added upstream in Mbed TLS, which have not yet been merged to master in Mbed OS. Neither has the int returning API. This PR must be somehow synchronized with the import of the newest upstream Mbed TLS code.

I agree that the codes should be agreed with the original developer. My intention was to return any reasonable error code from the already available ones while maintaining the information about the errors in the comments as it was provided in the first place.

I think that fine tuning the actual return codes (maybe expanding the error code space for the HW acceleration) may be a task big enough for a separate PR, however I am open to suggestions about changing the return codes in this PR.